File: docs-examples/example-java/src/test/java/io/micronaut/jms/docs/configuration/CustomBrokerSpec.java
package io.micronaut.jms.docs.configuration;

import io.micronaut.inject.qualifiers.Qualifiers;
import io.micronaut.jms.docs.AbstractJmsSpec;
import io.micronaut.jms.pool.JMSConnectionPool;
import io.micronaut.jms.pool.PooledConnection;
import jakarta.jms.ConnectionFactory;
import org.junit.jupiter.api.Test;

import jakarta.jms.Connection;
import jakarta.jms.JMSException;
import jakarta.jms.XAConnection;
import static org.junit.jupiter.api.Assertions.assertTrue;

class CustomBrokerSpec extends AbstractJmsSpec {

    @Test
    void testCustomBroker() throws JMSException {

        JMSConnectionPool connectionPool = applicationContext.getBean(
            JMSConnectionPool.class,
            Qualifiers.byName("activeMQXAConnectionFactory"));

        Connection connection = connectionPool.createConnection();
        assertTrue(connection instanceof PooledConnection);

        Connection realConnection = ((PooledConnection) connection).get();
        assertTrue(realConnection instanceof XAConnection);
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/jms/docs/configuration/CustomBrokerSpec.kt
package io.micronaut.jms.docs.configuration

import io.micronaut.inject.qualifiers.Qualifiers
import io.micronaut.jms.docs.AbstractJmsSpec
import io.micronaut.jms.pool.JMSConnectionPool
import io.micronaut.jms.pool.PooledConnection
import jakarta.jms.XAConnection
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class CustomBrokerSpec : AbstractJmsSpec() {

    @Test
    fun testCustomBroker() {
        val connectionPool = applicationContext.getBean(
            JMSConnectionPool::class.java,
            Qualifiers.byName("activeMQXAConnectionFactory")
        )

        val connection = connectionPool.createConnection()
        assertTrue(connection is PooledConnection)

        val realConnection = (connection as PooledConnection).get()
        assertTrue(realConnection is XAConnection)
    }
}
